### PR TITLE
Merge and deduplicate rule notices

### LINKED ISSUE #119: Investigate merging notices
In #87, `tslint-to-eslint-config` began logging rule notices, but there wasn't any infrastructure that took into account the possibility of having to merge notices. I have updated line 81 in `convertRules.ts` to just merge the two notice arrays, but we should investigate creating a merger for rule notices.

### PR DESCRIPTION
**Pull request by egiurleo**
Fixes #119

### CODE REVIEW

File: src/rules/convertRules.ts [hunk 1]
```diff
@@ -69,16 +73,15 @@ export const convertRules = (
...
-
                 converted.set(changes.ruleName, {
                     ...existingConversion,
                     ruleArguments: merger(
                         existingConversion.ruleArguments,
                         newConversion.ruleArguments,
                     ),
-                    notices: [...existingNotices, ...newNotices],
+                    notices: Array.from(
+                        new Set([...existingConversion.notices, ...newConversion.notices]),
```
Comment (egiurleo) [inline src/rules/convertRules.ts]: Let me know if this is better than it was previously (setting a default value for notices earlier on vs in the `existingNotices` and `newNotices` variables.

File: src/rules/convertRules.test.ts [hunk 2]
```diff
@@ -183,6 +184,51 @@ describe("convertRules", () => {
         );
     });
 
+    it("merges and deduplicates rule notices", () => {
```
Comment (egiurleo) [inline src/rules/convertRules.test.ts]: This test is mostly copy pasta. I don't think it has anything unnecessary but happy to remove stuff if I missed something :)

**Comment by egiurleo** [comment]
@JoshuaKGoldberg take a look :)

File: src/rules/convertRules.ts (hunk 1 above)
Comment (JoshuaKGoldberg) [reply src/rules/convertRules.ts]: @egiurleo unfortunately no, if `newNotices` is `undefined`, this will throw an error:

Trying out in a Node CLI:

```ts
> [...undefined, ...[]]
Thrown:
TypeError: undefined is not iterable (cannot read property Symbol(Symbol.iterator))
```

...which seems like a missing unit test!

File: src/rules/convertRules.test.ts (hunk 2 above)
Comment (JoshuaKGoldberg) [reply src/rules/convertRules.test.ts]: Yeah, I think this file has a lot of [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)/[WET](https://dev.to/wuz/stop-trying-to-be-so-dry-instead-write-everything-twice-wet-5g33) violations. Either the source function is doing too much and/or the unit tests should have common logic moved into helpers.

A few options for you here, all of which I have no issue with:
* Ignore this comment and continue eating the 🍝
* File a followup issue to investigate this
* In this PR, move shared logic from these unit tests into a function in this file
* In this PR, refactor some logic out of `convertRules` so edge cases can be directly unit tested - for example, the contents of the `for` loop.

Comment (egiurleo) [reply src/rules/convertRules.test.ts]: Gonna ignore for now and create a follow-up issue.

File: src/rules/convertRules.test.ts [hunk 3]
```diff
@@ -183,6 +183,96 @@ describe("convertRules", () => {
...
+        };
+        const conversionResult = {
+            rules: [
+                {
+                    ruleName: "eslint-rule-a",
+                    notices: undefined,
+                },
+                {
+                    ruleName: "eslint-rule-a",
+                    notices: undefined,
```
Comment (egiurleo) [inline src/rules/convertRules.test.ts]: @JoshuaKGoldberg is this the test case you're looking for? (Will make a follow-up PR to DRY out these tests)

Comment (JoshuaKGoldberg) [reply src/rules/convertRules.test.ts]: Yeah, this is perfect! 🙌

**Comment by JoshuaKGoldberg** [comment]
Released in `0.2.4`! ✨
